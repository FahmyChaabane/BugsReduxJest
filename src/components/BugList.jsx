import React from "react";
import { useDispatch, useSelector } from "react-redux";
import { startResolveBug, startAssignBugToUser } from "../actions/bugs";
import { getUnresolvedBugSelector } from "../selectors/bugs";

export const BugList = () => {
  const dispatch = useDispatch();
  const bugs = useSelector(getUnresolvedBugSelector);
  return (
    <ul>
      {bugs.map((bug) => (
        <li key={bug.id}>
          {bug.id}
          {bug.name}
          <button onClick={() => dispatch(startResolveBug(bug.id))}>
            resolve
          </button>
          <button onClick={() => dispatch(startAssignBugToUser(bug.id))}>
            assign to user
          </button>
        </li>
      ))}
    </ul>
  );
};

export default BugList;
