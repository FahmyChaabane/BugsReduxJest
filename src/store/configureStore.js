import { configureStore, getDefaultMiddleware } from "@reduxjs/toolkit";
import reducer from "../reducers/reducer";
import { logger } from "../middlewares/logger";

// eslint-disable-next-line
export default () =>
  configureStore({
    reducer,
    middleware: [...getDefaultMiddleware(), logger],
  });
