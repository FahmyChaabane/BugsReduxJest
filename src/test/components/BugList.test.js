import React from "react";
import { mount } from "enzyme";
import BugList from "../../components/BugList";
import configureStore from "../../store/configureStore";
import { Provider } from "react-redux";
const store = configureStore();

test("should correctly render BugList", () => {
  const wrapper = mount(
    <Provider store={store}>
      <BugList />
    </Provider>
  );
  expect(wrapper).toMatchSnapshot();
});
