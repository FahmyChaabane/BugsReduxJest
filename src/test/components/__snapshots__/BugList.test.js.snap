// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should correctly render BugList 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <BugList>
    <ul />
  </BugList>
</Provider>
`;
